internal/subscriber/retry/delayconsumer_test.go

```diff
@@ -81,7 +92,157 @@ func TestDelayConsumer_Run(t *testing.T) {
+	}{
+		{
+			name:    "Test Delay Consumer one message counsume out of 2",
+			wantErr: false,
+			args: args{
+				messages: msgs,
+			},
+			expected: msg2,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			count := 0
+			consumer.EXPECT().ReceiveMessages(gomock.Any(), gomock.Any()).DoAndReturn(
+				func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+					if count == 0 {
+						count++
+						return &messagebroker.GetMessagesFromTopicResponse{Messages: tt.args.messages}, nil
+					} else {
+						return &messagebroker.GetMessagesFromTopicResponse{}, nil
+					}
+				}).AnyTimes()
+			go dc.Run(ctx)
+			select {
+			case err := <-errCh:
+				if !tt.wantErr {
+					t.Errorf("Got Error : %v", err)
+				}
+				cancel()
+			case <-time.NewTicker(time.Millisecond * 500).C:
+				assert.Equal(t, len(dc.cachedMsgs), 1)
+				assert.True(t, reflect.DeepEqual(dc.cachedMsgs[0], tt.expected))
+				cancel()
+			}
+			<-time.NewTicker(time.Millisecond * 1).C
+		})
+	}
+}
+
+func TestDelayConsumer_Run_DeadLetter(t *testing.T) {
+	ctx, ctrl, subCtx, cancel, store, handler, consumer, cache := setupDC(t)
+	subs := getDummySubscriptionModel()
+	subscriberID := "subscriber-dl-id"
+	errCh := make(chan error)
+	dlqTopicChan := make(chan *messagebroker.SendMessageToTopicRequest)
+	producer := buildMockDLQProducer(ctx, ctrl, dlqTopicChan)
+	store.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+	dc, err := NewDelayConsumer(subCtx, subscriberID, "t1-dl", subs, store, handler, cache, errCh)
+	assert.NotNil(t, dc)
+	assert.Nil(t, err)
+	assert.NotNil(t, dc.LogFields())
+	tFut := time.Now().Add(time.Second * 10)
+	msgs := make([]messagebroker.ReceivedMessage, 0)
+	msg := getDummyBrokerMessageDLQ()
+	msg.NextDeliveryTime = tFut
+	msgs = append(msgs, msg)
+	count := 0
+	type args struct {
+		messages []messagebroker.ReceivedMessage
+	}
+	tests := []struct {
+		name     string
+		wantErr  bool
+		expected *messagebroker.SendMessageToTopicRequest
+		args     args
+	}{
+		{
+			name: "Test Delay Consumer msg pushed to DLQ success",
+			args: args{
+				messages: msgs,
+			},
+			expected: getMockSendMessageToTopicRequest(),
+			wantErr:  false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			consumer.EXPECT().ReceiveMessages(gomock.Any(), gomock.Any()).DoAndReturn(
+				func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+					if count == 0 {
+						count++
+						return &messagebroker.GetMessagesFromTopicResponse{Messages: tt.args.messages}, nil
+					} else {
+						return &messagebroker.GetMessagesFromTopicResponse{}, nil
+					}
+				}).AnyTimes()
+			go dc.Run(ctx)
+			for {
+				select {
+				case err := <-errCh:
+					t.Errorf("Got Error : %v", err)
+				case req := <-dlqTopicChan:
```

Comment:
since test case is to check if message is pushed to DLQ and input are also designed for that.Since purpose of test being fulfilled, why we need to spin up a consumer?